e2e/cypress/integration/login/tutorial_navigation_and_links_spec.js

```diff
@@ -0,0 +1,149 @@
+        //
+        // This fetches the contents of the appDownloadImage link and checks it without actually
+        // clicking it and opening it in a new tab as Cypress doesn't appear to handle this.
+
+        cy.get('#appDownloadImage').then(($a) => {
+            const href = $a.prop('href');
+
+            // request the contents of https://www.google.com/
+            cy.request(href).
+
+            // drill into the response body
+                its('body').
+                should('include', 'https://mattermost.com/wp-content/themes/mattermostv3/img/android-icon.png').
+                should('include', 'https://mattermost.com/wp-content/themes/mattermostv3/img/google-play.png').
+                should('include', 'https://mattermost.com/wp-content/themes/mattermostv3/img/ios.png').
+                should('include', 'https://mattermost.com/wp-content/themes/mattermostv3/img/app-store.png').
+                should('include', 'https://mattermost.com/wp-content/themes/mattermostv3/img/windows-icon.png');
+        });
+    }); // click link for new tab
+
+    it('Clicking the Off-Topic link in sidebar should display that channel highlighted in the sidebar.', () => {
+        cy.get('#sidebarItem_off-topic').parent().should('not.have.class', 'active');
+        cy.get('#sidebarItem_off-topic').click();
+        cy.get('#sidebarItem_off-topic').parent().should('have.class', 'active');
+
+        //  Retest tutorial to ensure it has not changed:
+        checkStepTwo();
+    }); // click Off Topic link in sidebar
+
+    it('Clicking Next on page 2 should lead to page 3.', () => {
+        cy.get('#tutorialNextButton').click();
+        checkStepThree();
+    }); // click next to see step 3
+
+    it('Clicking Next on step 3 should hide the tutorial and show the previously selected Off Topic channel', () => {
+        cy.get('#tutorialNextButton').click();
+        checkOffTopicChannel();
+    }); // click to see Off Topic channel
+
+    it('Another user who views the tutorial and clicks Skip button should go right to the default channel.', () => {
+        cy.apiLogout();
+        cy.loginAsNewUserWithTutorial();
+        checkStepOne();
+        cy.get('#tutorialSkipLink').click();
+        checkTownSquare();
+    });
+}); // describe
+
+/*  This function checks the contents of the first step of the tutorial.
+*/
+
+function checkStepOne() {
+    cy.get('#tutorialIntroContent').should('be.visible').should('contain', 'Welcome to:');
+    cy.get('#tutorialIntroContent').should('be.visible').should('contain', 'Mattermost');
+    cy.get('#tutorialIntroContent').should('be.visible').should('contain', 'Your team communication all in one place, instantly searchable and available anywhere.');
+    cy.get('#tutorialIntroContent').should('be.visible').should('contain', 'Keep your team connected to help them achieve what matters most.');
+    cy.get('#tutorialIntroCircle0').should('be.visible');
```

Comment:
We have so duplicate code in these 3 functions. Create another function for handling the shared element verifications for all three steps:

(Additionally, bonus test if you can check the active circle as well)

```suggestion
    function verifyCirclesAndFooter(activeStep) {
          cy.get('#tutorialIntroCircle0').should('be.visible');
          cy.get('#tutorialIntroCircle1').should('be.visible');
          cy.get('#tutorialIntroCircle2').should('be.visible');
          cy.get('#tutorialNextButton').should('be.visible');
          cy.get('#tutorialSkipLink').should('be.visible');
    }
```

Reply:
Created separate function for testing dots and links.
Added checks for 'has.class', 'circle active' on each step to ensure proper dot is highlighted.